src/wallets/sendTransaction/ConfirmAdvancedTransaction.tsx

```diff
@@ -0,0 +1,89 @@
+import React, {useState} from 'react'
+import {ModalProps} from 'antd/lib/modal'
+import {ModalLocker} from '../../common/MantisModal'
+import {Dialog} from '../../common/Dialog'
+import {DialogInput, DialogInputPassword} from '../../common/dialog/DialogInput'
+import {asEther} from '../../common/units'
+import {DialogShowAmount} from '../../common/dialog/DialogShowAmount'
+import {LoadedState} from '../../common/wallet-state'
+import {useTranslation} from '../../settings-state'
+import {withStatusGuard, PropsWithWalletState} from '../../common/wallet-status-guard'
+import {AdvancedTransactionParams} from './common'
+
+interface ConfirmAdvancedTransactionProps {
+  onClose: () => void
+  transactionParams: AdvancedTransactionParams
+  onCancel: () => void
+}
+
+const _ConfirmAdvancedTransaction = ({
+  transactionParams,
+  onCancel,
+  onClose,
+  walletState,
+}: PropsWithWalletState<
+  ConfirmAdvancedTransactionProps & ModalProps,
+  LoadedState
+>): JSX.Element => {
+  const {sendTransaction} = walletState
+  const {t} = useTranslation()
+  const modalLocker = ModalLocker.useContainer()
+
+  const {recipient, amount, gasLimit, gasPrice, data, nonce} = transactionParams
+  const [password, setPassword] = useState('')
+
+  return (
+    <>
+      <Dialog
+        title={t(['wallet', 'title', 'send'])}
+        leftButtonProps={{
+          children: t(['wallet', 'button', 'back']),
+          onClick: onCancel,
+          disabled: modalLocker.isLocked,
+        }}
+        rightButtonProps={{
+          children: t(['wallet', 'button', 'confirm']),
+          onClick: (): void => {
+            sendTransaction({
+              recipient,
+              gasLimit: parseInt(gasLimit),
+              gasPrice: asEther(gasPrice),
+              data,
```

Comment:
What format should I use when interacting with a contract?

Validation and a working example would be nice.
e.g. deploying a contract and calling it.

Reply:
Added relevant validation for all fields, please check if it makes sense :)